_search-plugins/concurrent-segment-search.md

```diff
@@ -0,0 +1,162 @@
+```
+{% include copy.html %}
+
+If you want to define the environment variable separately, prior to running OpenSearch, run the following commands:
+
+```bash
+export OPENSEARCH_JAVA_OPTS="-Dopensearch.experimental.feature.concurrent_segment_search.enabled=true"
+```
+{% include copy.html %}
+
+```bash
+./bin/opensearch
+```
+{% include copy.html %}
+
+## Disabling concurrent search at index or cluster level
+
+After you enable the experimental feature flag, all search requests will use concurrent segment search during query phase. To disable concurrent segment search for all indexes, set following dynamic cluster setting:
+
+```json
+PUT _cluster/settings
+{
+   "persistent":{
+      "search.concurrent_segment_search.enabled": false
+   }
+}
+```
+{% include copy-curl.html %}
+
+To disable concurrent segment search for a particular index, specify the index name in the endpoint:
+
+```json
+PUT <index-name>/_settings
+{
+    "index.search.concurrent_segment_search.enabled": false
+}
+```
+{% include copy-curl.html %}
+
+## Slicing mechanisms
+
+You can choose one of the two available mechanisms of assigning segments to slices: the default [Lucene mechanism](#the-lucene-mechanism) or the [max slice count mechanism](#the-max-slice-count-mechanism).
+
+### The Lucene mechanism
+
+By default, Lucene assigns a maximum of 250K documents or 5 segments (whichever is met first) to each slice in a shard. For example, consider a shard with 11 segments. The first 5 segments have 250K documents each and the next 6 segments have 20K documents each. The first 5 segments will be assigned to one slice each because they each contain the maximum allowed document count for a slice. Then the next 5 segments will all be assigned to another single slice because of the maximum allowed segment count for a slice. The 11th slice will be assigned to a separate slice. 
+
+### The max slice count mechanism
+
+The _max slice count_ mechanism is an alternative slicing mechanism that uses a statically configured maximum number of slices and divides segments among the slices in a round-robin fashion. This is useful when there are already too many top-level shard requests and you want to limit the number of slices per request in order to reduce competition between the slices.
+
+### Setting the slicing mechanism
+
+By default, concurrent segment search uses the Lucene mechanism to calculate the number of slices for each shard-level request. To use the max slice count mechanism instead, configure the `search.concurrent.max_slice_count` static setting in the `opensearch.yml` config file:
+
+```yaml
+search.concurrent.max_slice_count: 2
+```
+{% include copy.html %}
+
+The `search.concurrent.max_slice_count` setting can take the following valid values:
+- `0`: Use the default Lucene mechanism.
+- Positive integer: Use the max target slice count mechanism. Usually, a value 2--8 should be sufficient.
```

Comment:
Would it be accurate to instead say "a value between 2 and 8"? There's technically nothing wrong with this, but it's not how it would normally be expressed.